spring-cloud-task-docs/src/main/asciidoc/appendix-cloud-foundry.adoc

```diff
@@ -0,0 +1,34 @@
+
+[[appendix-cloud-foundry]]
+== Running a Task App on Cloud Foundry
+
+This section will discuss how to deploy a Spring Cloud Task application onto
+the Cloud Foundry platform.
+
+There are 3 steps to deploying your Spring Cloud Task application:
+
+. We must push your application to the platform using the `cf push command`.
+This command pushes your jar to the Cloud Foundry platform.
+
+. We need to stop the application from running once the push is complete.
+Cloud Foundry has to run the application to stage it.  But when Cloud Foundry
+runs the application at this time it is treating the application as a LRP (Long Running Process)
+which means once the task is complete it will restart it.  This is not the desired
+behavior for our task, thus we must stop it from running.  To stop the LRP execution,
+we will use the `cf stop` command.
+
+. We now need to launch our application as a task to do this we will use
+the `cf run-task` command.
+
+NOTE: When using the run-task command you have to specify build pack to be used
+and that is specified in the `command string` enumerated below.
+
+[source,bash]
+```
+$ cf push --health-check-type none -p <location of your jar> <name of your app> --no-route
+$ cf stop <name of your app>
+$ cf run-task <name of your app> <command string>
```

Comment:
I think there is more to it here given 
```cf run-task message-producer ".java-buildpack/open_jdk_jre/bin/java org.springframework.boot.loader.JarLauncher --name messages```

there is `--name` and also the `<command string>` is not something one can easily figure out.

Reply:
If you don't know the command up front, you have to run the app once just to get the command.. We should list the command here, e.g. 

```
$ cf run-task simpletask-mlp --name simpletask-1-mlp 'JAVA_OPTS="-agentpath:$PWD/.java-buildpack/open_jdk_jre/bin/jvmkill-1.12.0_RELEASE=printHeapHistogram=1 -Djava.io.tmpdir=$TMPDIR -Djava.ext.dirs=$PWD/.java-buildpack/container_security_provider:$PWD/.java-buildpack/open_jdk_jre/lib/ext -Djava.security.properties=$PWD/.java-buildpack/java_security/java.security $JAVA_OPTS" && CALCULATED_MEMORY=$($PWD/.java-buildpack/open_jdk_jre/bin/java-buildpack-memory-calculator-3.13.0_RELEASE -totMemory=$MEMORY_LIMIT -loadedClasses=12876 -poolType=metaspace -stackThreads=250 -vmOptions="$JAVA_OPTS") && echo JVM Memory Configuration: $CALCULATED_MEMORY && JAVA_OPTS="$JAVA_OPTS $CALCULATED_MEMORY" && MALLOC_ARENA_MAX=2 SERVER_PORT=$PORT eval exec $PWD/.java-buildpack/open_jdk_jre/bin/java $JAVA_OPTS -cp $PWD/. org.springframework.boot.loader.JarLauncher'
```

Nothing seems that specific to the app, so other than some version numbers in there, it looks like boilerplate.

Reply:
Hmm, the command args are not quite boiler plate.  Trying that command line on other instances of CF, caused the task to fail.